File: 프로그래머스/lv2/42584. 주식가격/주식가격.py
def solution(prices):
    
    cnt = 0
    sec = []
    
    for i in range(len(prices)):
        for j in range(i+1, len(prices)):
            if prices[i] <= prices[j]:
                cnt += 1
            else: 
                cnt += 1
                break
        sec.append(cnt)
        cnt = 0
    
    return sec

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;
class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        Stack<Integer> s = new Stack<>();
        for (int i = 0; i < prices.length-1;i++){
            for (int j = i + 1; j < prices.length; j++){
                if (prices[i] <= prices[j]) {
                    s.add(prices[j]);
                } else {
                    s.add(prices[j]);
                    answer[i] = s.size() ;
                    s.clear();
                    break;
                }
                if (j == prices.length -1) {
                    answer[i] = s.size() ;
                    s.clear();
                }
            }
        }
        return answer;
    }
}